presto-main/src/main/java/com/facebook/presto/operator/spiller/BinarySpillerFactory.java

```diff
@@ -33,28 +33,27 @@
 public class BinarySpillerFactory
         implements SpillerFactory
 {
-    public static final Path SPILL_PATH = Paths.get("/tmp/spills");
-
     private final ListeningExecutorService executor;
     private final BlockEncodingSerde blockEncodingSerde;
+    private final Path spillPath;
 
     @Inject
-    public BinarySpillerFactory(BlockEncodingSerde blockEncodingSerde)
+    public BinarySpillerFactory(BlockEncodingSerde blockEncodingSerde, FeaturesConfig featuresConfig)
     {
-        this(blockEncodingSerde, MoreExecutors.listeningDecorator(newFixedThreadPool(4, daemonThreadsNamed("binary-spiller-%s"))));
+        this(blockEncodingSerde, MoreExecutors.listeningDecorator(newFixedThreadPool(4, daemonThreadsNamed("binary-spiller-%s"))), featuresConfig);
     }
 
-    public BinarySpillerFactory(BlockEncodingSerde blockEncodingSerde, ListeningExecutorService executor)
+    public BinarySpillerFactory(BlockEncodingSerde blockEncodingSerde, ListeningExecutorService executor, FeaturesConfig featuresConfig)
     {
         this.blockEncodingSerde = requireNonNull(blockEncodingSerde, "blockEncodingSerde is null");
         this.executor = requireNonNull(executor, "executor is null");
-
-        SPILL_PATH.toFile().mkdirs();
+        this.spillPath = featuresConfig.getSpillerSpillPath();
+        this.spillPath.toFile().mkdirs();
     }
 
     @Override
     public Spiller create(List<Type> types)
     {
-        return new BinaryFileSpiller(blockEncodingSerde, executor);
+        return new BinaryFileSpiller(blockEncodingSerde, executor, spillPath);
```

Comment:
I think each BinaryFileSpiler should get an unique spill path. In case where are two spilling aggregation running concurrently then they might override spills to each other.